TFT/src/User/API/menu.h

```diff
@@ -52,7 +52,8 @@ typedef enum
   MENU_TYPE_LISTVIEW,
   MENU_TYPE_DIALOG,
   MENU_TYPE_FULLSCREEN,
-  MENU_TYPE_OTHER
+  MENU_TYPE_OTHER,
+  MENU_TYPE_CUSTOM
```

Comment:
the MENU_TYPE_OTHER is already the same as MENU_TYPE_CUSTOM
```suggestion
  MENU_TYPE_OTHER
```

Reply:
Yes, I saw that but it seems it's not meant for the same usage. With the custom type I made just few simple extensions in menu.c and notification.c just to allow a custom redraw callback invoketion. Those changes are fully backward compatible with the existing logic. With type OTHER I don't see the possibility to invoke a custom redraw. Please have a look to the changes I made in menu.c and notification.c. With the custom type I can use the loopProcess(). When a temporary displayed info has to be cleared, the registered custom redraw callback is invoked to repaint the custom menu (Mesh editor).

Also it seems that other users (such as ETE-Design above) have the same problem I mentioned in your PR. At the end of "Updating Configuration" (for config.ini) the process stucks when storing the settings in the FLASH.

Reply:
@digant73 Is there a Work Around for the part where it stuck?

Reply:
> @digant73 Is there a Work Around for the part where it stuck?

Not yet. I simply commnented out the code line where the settings are stored to FLASH. But clearly this is not a final solution. I should try to move the store invoketion alsewhere to verify if it helps to avoid the freeze

Reply:
@digant73 What line to comment out? Would like to try your PR out :-)

Reply:
@ETE-Design comment out line 65 on config.c. See below

//    storePara();

Reply:
@guruathwal found the problem during the configuration update process. The current size for the data buffer used in 
 flashStore.c is PARA_SIZE 512 while the buffer requires at least 516 now. Can we set PARA_SIZE to 768 or any other bigger value than 512?

Reply:
@digant73  you do not need `softwareEndstops` variable in settings when you are polling it every time system boots.
the settings are stores on the main memory of the MCU. increasing PARA_SIZE will reduce the program-memory.
The colors for mesh grid can be hard codes to reduce the ram usage also.

Reply:
@guruathwal I preferred to make the colors configurable just because everyone can have problems with some colors instead of others. The RAM usage is about 34% now. In that sense it should not be a problem at all to change from 512 to 768.
I defined softwareEdnstops on machine settings as it was done for other machine settings such as babyStepping, promptSupport etc... They all are set on parseAck() and polled elsewhere.
Please also have a look to "void sizecheck(void)" in SanityCheck.c. It should be responsible to check the limits also for PARA_SIZE..,. but that function is never executed. Also the checks seems not correct. It should be better that those important checks are performed although maybe it's not possible at compile time.
In the next update, the usual improvements and optimization can always be provided